fe/main.go

```diff
@@ -206,8 +206,12 @@ func listServicesFromK8sAPI(serviceOwner string) (map[string]serviceData, error)
 // We want to get the time that best reflects the current age
 // of the container running the service.
 func bestStartTime(pod v1.Pod) time.Time {
-	podStartTime := pod.Status.StartTime.Time
-	bestStartTime := podStartTime
+	var bestStartTime time.Time
+
+	podStartTime := pod.Status.StartTime
+	if podStartTime != nil {
+		bestStartTime = podStartTime.Time
+	}
 
 	for _, cStatus := range pod.Status.ContainerStatuses {
```

Comment:
so the problem was `pod.Status.StartTime`? and we expect `pod.Status` to be non-nil even if the pod hasn't come up yet? (alternately, if pod hasn't come up, why do we bother with the rest of the code below -- what information do we realistically stand to get?)

Reply:
pod.Status can't be nil, it's a struct field

Reply:
and as far as "why do we bother with the rest of the code below?" eh, this server segfault is pretty easy to reproduce. if you have a reproducible case that you're worried about, we can address that too!